components/ScriptLibrary/include/LHVMTypes.h

```diff
@@ -0,0 +1,371 @@
+/******************************************************************************
+ * Copyright (c) 2018-2024 openblack developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/openblack/openblack
+ *
+ * openblack is licensed under the GNU General Public License version 3.
+ *******************************************************************************/
+
+#pragma once
+
+#include <cassert>
+#include <cstdint>
+
+#include <array>
+#include <functional>
+#include <map>
+#include <string>
+
+namespace openblack::lhvm
+{
+
+static const std::array<std::string, 31> k_OpcodeNames = {
+    "END",    "JZ",   "PUSH", "POP",       "ADD",       "SYS",        "SUB",       "NEG",  "MUL", "DIV", "MOD",
+    "NOT",    "AND",  "OR",   "EQ",        "NE",        "GE",         "LE",        "GT",   "LT",  "JMP", "SLEEP",
+    "EXCEPT", "CAST", "RUN",  "ENDEXCEPT", "RETEXCEPT", "FAILEXCEPT", "BRKEXCEPT", "SWAP", "LINE"};
+
+enum class LHVMVersion : uint32_t
+{
+	BlackAndWhite = 7,
+	CreatureIsle = 8,
+	BlackAndWhite2 = 12,
+
+	Last = 0xFFFFFFFF
```

Comment:
If I don't code a value high enough, the checks complain about the enum class underlying type (uint32_t) being too large for no reason. But the reason is I want `LHVMVersion` to be a 32 bit type, this way I can declare the `version` field of that type and size.
Having the field sized properly allows to use it directly as buffer when reading from/writing to file, avoiding the effort of using an additional variable to use as buffer of the proper size. Other than being verbose in source code, this could be a bit inefficient for types that are read thousands times per file, such as `DataType`, `VMMode` and `Opcode`, which are part of `VMInstruction`.

Reply:
Ah yes. I think the better thing is to add `-performance-enum-size,` after line 10 in `.clang-tidy` to disable the check.